pkg/flow/collector.go

```diff
@@ -0,0 +1,215 @@
+package flow
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/skupperproject/skupper/pkg/messaging"
+)
+
+type senderDirect struct {
+	sender    *sender
+	outgoing  chan interface{}
+	heartbeat bool
+}
+
+type ApiRequest struct {
+	RecordType int
+	ItemType   int
+	Request    *http.Request
+}
+
+type ApiResponse struct {
+	Body   *string
+	Status int
+}
+
+type eventSource struct {
+	Identity    *string       `json:"identity,omitempty"`
+	Beacon      *BeaconRecord `json:"beacon,omitempty"`
+	CurrentTime uint64        `json:"currentTime,omitempty"`
+	Heartbeats  int           `json:"heartbeats,omitempty"`
```

Comment:
Like the BEACON message, the HEARTBEAT message is also sent periodically by each event source.  Unlike the BEACON message, the HEARTBEAT message is sent to the source-specific multicast address so that it is only received by collectors that have explicitly subscribed to events from that source.